### PR TITLE
Update TOC styles through wrapped styled component

### PR DESCRIPTION
**Pull request by tdgao**
Updating table of contents

- [x] Documentation content
- [ ] Page Layout / templates / structure
- [ ] Internal / code cleanup / build system

### CODE REVIEW

File: features/TOCSidebar.feature [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        When they hover over a link in TOC
+        Then the item's text is in bold
+        And the item's left border is green #007A5B
+
+    @motivating
+    Scenario: Introduction section appears on top of TOC
+        Given a user is viewing a docs article with TOC
+        Then the first link in the TOC is "Introduction"
+        When they click on "Introduction"
+        Then the page scrolls to the first heading
```
Comment (DerekSiemens) [inline features/TOCSidebar.feature]:
```suggestion
        Then the page scrolls to the page heading
```

File: features/TOCSidebar.feature [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        Then only the nested link is in the hover state
+        When they click the link
+        Then the page scrolls to the corresponding sub-section
+        Examples:
+            | article                                |
+            | /success/referral-program-optimization |
+            | /developer/segment                     |
+
+    @ui
+    Scenario Outline: Table of content scrolls when overflowing from view
```
Comment (DerekSiemens) [inline features/TOCSidebar.feature]: If you only have on example, scenario outline is not needed

File: features/TOCSidebar.feature [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        Given a user is viewing a docs <article> with TOC
+        And the TOC is overflowing from view
+        Then they are able to scroll the TOC to access hidden links
+        Examples: 
+            | article                 |
+            | /designer/widget-editor |
+
+    @landmine
+    Scenario: Clicking on a link in TOC changes spacing in navigation bar
```
Comment (DerekSiemens) [inline features/TOCSidebar.feature:69]: I wonder if we can fix this quick, might be worth trying to fix quick when we are in the code

File: features/TOCSidebar.feature [hunk 4]
```diff
@@ -0,0 +1,60 @@
+Feature: Table of Contents
```
Comment (DerekSiemens) [inline features/TOCSidebar.feature:3]: Add author and owner tags at the top of the file

File: features/TOCSidebar.feature [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        Given a user is viewing a docs article with TOC
+        And the page's content has <headings>
+        Then the TOC has nested links
+        When they hover over a nested link
+        Then only the nested link is in the hover state
+        When they click the nested link
+        Then the page scrolls to the corresponding sub-section
+        Examples:
+            | headings    |
+            | h3, h4      |
```
Comment (DerekSiemens) [inline features/TOCSidebar.feature]:
```suggestion
            | h1, h2, h3, h4, h5, h6      |
```

File: features/TOCSidebar.feature [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            | h3, h4, h5  |
```
Comment (DerekSiemens) [inline features/TOCSidebar.feature]:
```suggestion
            | h2, h4, h5  |
```

File: src/components/TocFrame.tsx [hunk 7]
```diff
@@ -23,6 +25,54 @@ export type Entry = {
...
+  }
+  li > ul{
+    margin-top: 5px
+    margin-left: 12px
+  }
+  li{
+    list-style-type: none;
+    line-style:none;
+
+    border-left: 1px solid #E2E2E2;
```
Comment (zachharrison) [inline src/components/TocFrame.tsx]:
```suggestion
    border-left: 1px solid var(--sq-action-primary-disabled);
```

File: src/components/TocFrame.tsx [hunk 8]
```diff
@@ -23,6 +25,53 @@ export type Entry = {
...
+    padding: var(--sq-spacing-xx-small) var(--sq-spacing-small);
+
+    font-family: 'Helvetica';
+    font-style: normal;
+    font-weight: var(--sq-font-weight-regular);
+    font-size: var(--sq-font-size-regular);
+    color: var(--sq-text);
+  }
+  a:hover {
+    font-weight: 700;
```
Comment (zachharrison) [inline src/components/TocFrame.tsx]:
```suggestion
    font-weight: var( --sq-font-weight-bold);
```

**Review by zachharrison — DISMISSED** [review]

File: src/components/TocFrame.tsx [hunk 9]
```diff
@@ -34,6 +84,7 @@ export default function render({
 }) {
   return (
     <StickyContainer>
+      {/* {console.log("entry",entry)} */}
```
Comment (DerekSiemens) [inline src/components/TocFrame.tsx]: Can likely just remove this

File: src/assets/stylesheets/_misc.less [hunk 10]
```diff
@@ -150,7 +150,8 @@
   font-weight: bold;
   color: #999;
   margin-bottom: 10px;
-  border-bottom: 1px solid #ccc;
+  margin-top: 10px;
+  // border-bottom: 1px solid #ccc;
```
Comment (00salmon) [inline src/assets/stylesheets/_misc.less]:
```suggestion
```

**Review by 00salmon — COMMENTED** [review]
looks good to me 👍 `TOC.tsx` looks like it might need prettier run on it is all

**Review by 00salmon — APPROVED** [review]
🚀

**Pull request merged**